core/src/main/scala/org/elasticmq/actor/queue/QueueActorMessageOps.scala

```diff
@@ -105,36 +106,44 @@ trait QueueActorMessageOps extends Logging {
 
   protected def receiveMessages(visibilityTimeout: VisibilityTimeout,
                                 count: Int,
-                                receiveRequestAttemptId: Option[String]): List[MessageData] = {
-    implicit val np = nowProvider
-    val messages = receiveRequestAttemptId
-      .flatMap({ attemptId =>
-        // for a given request id, check for any messages we've dequeued and cached
-        val cachedMessages = getMessagesFromRequestAttemptCache(attemptId)
-
-        // if the cache returns an empty list instead of None, we still want to pull messages from
-        // from the queue so return None in that case to properly process down stream
-        cachedMessages.getOrElse(Nil) match {
-          case Nil     => None
-          case default => Some(default)
+                                receiveRequestAttemptId: Option[String]): Either[ElasticMQError, List[MessageData]] = {
+    if (inflightMessagesRegisty.size >= queueData.inflightMessagesLimit)
+      Left(new OverLimitError(queueData.name))
+    else {
+      implicit val np = nowProvider
```

Comment:
maybe extract this to a method?

Reply:
I did refactor receiving messages to private method